File: examples/csharp/http-client/TablesClient.cs
using System.Net;
using System.Net.Http.Headers;

namespace CdcExamples.Http;

/// <summary>
/// TablesClient ruft die Tabellen-Auflistung mit dem <c>reader</c>-Token ab
/// (<c>LH-FA-SST-006</c>, <c>ADR-0057</c>) — Form-Vorbild:
/// <c>examples/http-client/main.go</c>, <c>listTables</c>. Ein Nicht-200-
/// Status ist ein sichtbarer Fehler, kein stiller Leerwert. Der
/// <see cref="System.Net.Http.HttpClient"/> ist injiziert, damit der
/// Fehlerpfad ohne erreichbaren Host netzlos testbar ist.
/// </summary>
public static class TablesClient
{
    public static async Task<string> ListTablesAsync(
        System.Net.Http.HttpClient httpClient,
        Config cfg,
        CancellationToken cancellationToken = default)
    {
        var url = TablesUrlBuilder.Build(cfg.Addr, cfg.Source, cfg.Publication);
        using var request = new HttpRequestMessage(HttpMethod.Get, url);
        request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", cfg.Token);

        using var response = await httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
        var body = await response.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false);
        if (response.StatusCode != HttpStatusCode.OK)
        {
            throw new InvalidOperationException($"HTTP-Status {(int)response.StatusCode}: {body}");
        }
        return body;
    }
}

File: examples/kotlin/http-client/src/main/kotlin/cdcexamples/http/TablesClient.kt
package cdcexamples.http

import java.net.URI
import java.net.http.HttpClient
import java.net.http.HttpRequest
import java.net.http.HttpResponse
import java.time.Duration

/**
 * TablesClient ruft die Tabellen-Auflistung mit dem `reader`-Token ab
 * (`LH-FA-SST-006`, `ADR-0057`) — Form-Vorbild:
 * `examples/http-client/main.go`, `listTables`. Ein Nicht-200-Status ist ein
 * sichtbarer Fehler, kein stiller Leerwert. Der [HttpClient] ist injiziert,
 * damit der Fehlerpfad ohne erreichbaren Host netzlos testbar ist.
 */
object TablesClient {
    private val requestTimeout: Duration = Duration.ofSeconds(10)

    fun listTables(httpClient: HttpClient, cfg: Config): String {
        val url = TablesUrlBuilder.build(cfg.addr, cfg.source, cfg.publication)
        val request = HttpRequest.newBuilder(URI.create(url))
            .header("Authorization", "Bearer ${cfg.token}")
            .timeout(requestTimeout)
            .GET()
            .build()

        val response = httpClient.send(request, HttpResponse.BodyHandlers.ofString())
        if (response.statusCode() != 200) {
            throw IllegalStateException("HTTP-Status ${response.statusCode()}: ${response.body()}")
        }
        return response.body()
    }
}
